webrender/src/render_backend.rs

```diff
@@ -105,6 +105,11 @@ struct Document {
     /// Properties that are resolved during frame building and can be changed at any time
     /// without requiring the scene to be re-built.
     dynamic_properties: SceneProperties,
+
+    /// Track whether the last built frame is up to date or if it will need to be re-built
+    /// before rendering again.
+    frame_is_valid: bool,
+    hit_tester_is_valid: bool,
```

Comment:
if it's not valid, can we just have `hit_tester == None`?

Reply:
If we do this we need to first process all messages that use the hit tester before setting it to `None`. for example if you have a transaction that both updates and uses the hit tester, right now the rule is that hit testing operations use the previous hit tester and we update the hit tester at the end.
So it'll be a bit more fiddly but I think it's a good idea.

Reply:
Alternatively, I'm leaning towards replacing the `Option<HitTester>` with a bare `HitTester` containing a valid boolean flag. This would let us recycle it's allocations instead of reallocating each time. Thoughts?

Reply:
everything for performance! :)